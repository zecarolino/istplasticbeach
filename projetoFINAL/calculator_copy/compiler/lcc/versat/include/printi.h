#ifndef _PRINTI_H_
#define _PRINTI_H_

#include "mod.h"
#include "div.h"
#include "putchar.h"

void printi(int n, int b) {
	int a, c;
	if (a = n / b) printi(a, b);
	c = n % b;
	if (c <= 9) putchar(c + '0');
	else putchar(c + 'A' -10);
}

#endif /* _PRINTI_H_ */
