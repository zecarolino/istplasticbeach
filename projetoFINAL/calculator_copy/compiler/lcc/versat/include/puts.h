#ifndef _PUTS_H_
#define _PUTS_H_

#include "putchar.h"

int puts(char *s) {
	while (*s) putchar(*s++);
	return 0;
}

#endif /* _PUTS_H_ */
