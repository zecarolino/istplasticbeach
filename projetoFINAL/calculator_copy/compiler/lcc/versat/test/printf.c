#include "printi.h"
#include "puts.h"

typedef int *va_list;
#define va_start(ap,fmt) ((ap)=(va_list)&(fmt))
#define va_arg(ap,type) ((type)*++ap)
#define va_end(ap)

int printf(const char *fmt, ...)
{
  va_list ap;
  va_start(ap, fmt);
  while (*fmt != 0) {
    if (*fmt == '%') {
      fmt++;
      if (*fmt == 'd') printi(va_arg(ap,int), 10);
      else if (*fmt == 's') puts(va_arg(ap,char*));
      else if (*fmt == 'c') putchar(va_arg(ap,int));
      else putchar(*fmt);
    }
    else putchar(*fmt);
    fmt++;
  }
  va_end(ap);
  return 1;
}

int main() {
	printf("Hell%c w%s!\n642=%d\n", 'o', "orld", 642);
	return 0;
}
