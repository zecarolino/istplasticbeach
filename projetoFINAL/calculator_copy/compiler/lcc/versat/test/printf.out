VCD info: dumpfile xtop.vcd opened for output.
Hello world!
642=642
0
Catched Trap at time  7094 clock cycles:
